src/linear_algebra/linear_independent.lean

```diff
@@ -984,3 +984,14 @@ have finite s, from u.finite_to_set.subset hsu,
 ⟨this, by rw [←eq]; exact (finset.card_le_of_subset $ finset.coe_subset.mp $ by simp [hsu])⟩
 
 end vector_space
+
+lemma linear_independent.restrict_scalars {R S M ι : Type*} [semiring R] [semiring S]
```

Comment:
Is there a section above with the submodule assumptions already?

Reply:
You are right: I moved the lemma to `section semimodule` and recycled as many `variables` as I could in the statement.  I could not find a place where there were two different `semirings`.  I wonder whether I should have started such a section, but decided against it: if there were more results along these lines, maybe it would make sense to split the file and follow the "standard topology approach": properties of a single "point" in one file, properties about relations between different "points" in separate files.